### PR TITLE
test(e2e): use ZOT as testing backend for `oras blob` specs

### PR DESCRIPTION
**Pull request by qweeah**
**What this PR does / why we need it**:
This PR uses ZOT as the testing backend for `oras blob` E2E specs, depends on #1081

### CODE REVIEW

File: test/e2e/suite/command/blob.go [hunk 1]
```diff
@@ -169,28 +171,22 @@ var _ = Describe("1.1 registry users:", func() {
...
+			CopyZOTRepo(BlobRepo, dstRepo)
+			toDeleteRef := RegistryRef(ZOTHost, dstRepo, foobar.FooBlobDigest)
 			ORAS("blob", "delete", toDeleteRef).
 				WithInput(strings.NewReader("y")).
 				MatchKeyWords("Deleted", toDeleteRef).Exec()
-			ORAS("blob", "delete", toDeleteRef).
-				WithDescription("validate").
-				WithInput(strings.NewReader("y")).
-				ExpectFailure().
-				MatchErrKeyWords("Error:", toDeleteRef, "the specified blob does not exist").Exec()
```
Comment (wangxiaoxuan273) [inline test/e2e/suite/command/blob.go:181]: Why remove this `validate` part? What happens on Zot if we delete the blob again?

Comment (qweeah) [reply test/e2e/suite/command/blob.go:181]: ZOT's implementation if not like distribution, it returns 202 for deletion, which mean the deletion will be done asynchronizely. The blob can still be fetched and re-delete after being deleted :)

I raised a query in ZOT's slack channel and waiting for their confirm https://cloud-native.slack.com/archives/C03EGRE4QGH/p1693210694637909

Comment (wangxiaoxuan273) [reply test/e2e/suite/command/blob.go:181]: So if I delete a blob and **immediately** fetch it, the deletion may not yet be completed and I may still get it? And if I fetch it after a day, the deletion may be done and I'll get 404? Is it right?

Comment (qweeah) [reply test/e2e/suite/command/blob.go:181]: Sounds strange but yes. I am playing with the GC configuration but haven't find a way to instantly reflect a blob deletion yet.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**